hedera-node/hedera-schedule-service/src/main/java/com/hedera/node/app/service/schedule/ScheduleService.java

```diff
@@ -58,4 +66,26 @@ default Set<RpcServiceDefinition> rpcDefinitions() {
     static ScheduleService getInstance() {
         return RpcServiceFactory.loadService(ScheduleService.class, ServiceLoader.load(ScheduleService.class));
     }
+
+    /**
+     * An executable transaction with the verifier to use for child signature verifications. If set,
+     * "not before" (nbf) time is the earliest consensus time at which the transaction could be executed.
+     */
+    record ExecutableTxn(
+            TransactionBody body,
+            VerificationAssistant verificationAssistant,
+            AccountID payerId,
+            @Nullable Instant nbf) {}
+
+    /**
+     * Given a [start, end) interval and a supplier of a StoreFactory that can be used in the returned
+     * iterator's remove() implementation to get a StoreFactory to purge a successfully executed txn,
+     *
+     * @return an iterator over all ExecutableTxn this service wants to execute in the interval.
+     */
+    default Iterator<ExecutableTxn> iterTxnsForInterval(
+            Instant start, Instant end, Supplier<StoreFactory> cleanupStoreFactory) {
+        // Default implementation returns an empty iterator
+        return null;
```

Comment:
Returning `Collections.emptyIterator()` is likely a safer default.

Reply:
Done.